core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java

```diff
@@ -355,62 +357,52 @@ public static class AuthSession {
     private static final long MAX_REFRESH_WINDOW_MILLIS = 300_000; // 5 minutes
     private static final long MIN_REFRESH_WAIT_MILLIS = 10;
     private volatile Map<String, String> headers;
-    private volatile String token;
-    private volatile String tokenType;
-    private volatile Long expiresAtMillis;
-    private final String credential;
-    private final String scope;
-    private volatile boolean keepRefreshed = true;
+    private volatile AuthConfig config;
 
     /**
-     * @deprecated will be removed in 1.3.0; use {@link AuthSession#AuthSession(Map, String, String,
-     *     String, String)} instead.
+     * @deprecated will be removed in 1.3.0; use {@link AuthSession#AuthSession(Map, AuthConfig)}
+     *     instead.
      */
     @Deprecated
     public AuthSession(Map<String, String> baseHeaders, String token, String tokenType) {
-      this(baseHeaders, token, tokenType, null, OAuth2Properties.CATALOG_SCOPE);
+      this(baseHeaders, ImmutableAuthConfig.builder().token(token).tokenType(tokenType).build());
```

Comment:
This seems like a ton of changes just to disable refresh. Why not just call `stopRefreshing()` after creating the `AuthSession` and avoid all these changes?

Reply:
I considered that, but `keepRefreshed` is not carried over from one AuthSession to the other, thus being more error prone and difficult to understand when exactly `keepRefreshed` would have to be set. 
We would end up having to carry over the `keepRefreshed` flag from one AuthSession to the next and that would require updating the constructor params. 
I think long term having an `AuthConfig` makes it easier to reason about the configuration that is set.

Reply:
Actually, I think another idea is to pass a null executor. That way if refresh is disabled, we don't start the thread and never attempt to schedule. Then we don't need these big changes.

Reply:
using a null executor was the very first approach I had to this (and I made it available in https://github.com/apache/iceberg/pull/6867) but I discarded it after reviewing and thinking about it. 

While the diff in https://github.com/apache/iceberg/pull/6867 ends up being smaller, the code is still somewhat difficult to grasp if you're reading it.
One has to pass through all the `fromXyz()` methods to really understand what's happening with the parameters that are being passed around.

Having an `AuthConfig` is fairly straight-forward to understand and one doesn't have to go through multiple layers of methods to figure out how the `AuthSession` ends up being configured.

Obviously during PR reviews we only see a diff and a smaller diff is always easier to review. However, sometimes it's better to produce a slightly bigger diff to improve the code for readability and understandability.

Reply:
looks like we are still settling this debate?

To me, overall using `AuthConfig` does seem cleaner, but why not just make `AuthSession` immutable in that case?

Reply:
unfortunately we can't make `AuthSession` itself immutable because it modifies its own state during [token refresh](https://github.com/apache/iceberg/blob/28e419d5bc9420eb86298ba9b834763b250c677a/core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java#L428-L470)

Reply:
I see, I was thinking to just replace the AuthSession at one level above during refresh, but maybe that's too much complexity.

Reply:
I prefer the other set of changes. It seems better to me to not start a refresh thread and to then skip refreshing, than to configure each auth session individually with a global setting.

I also don't think it makes code more readable to introduce an additional `AuthConfig` interface and implementation.

Reply:
I have pushed a version of the code that just passes a null executor to indicate no token refresh scheduling